runtime/onert/backend/train/ExtraTensorGenerator.cc

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "ExtraTensorGenerator.h"
+
+#include "ExtraTensorIndex.h"
+
+#include <ir/Operations.h>
+#include <util/logging.h>
+#include <memory>
+
+namespace onert
+{
+namespace backend
+{
+namespace train
+{
+
+ExtraTensorGenerator::ExtraTensorGenerator(const ir::train::TrainableGraph *tgraph,
+                                           std::shared_ptr<TensorBuilder> &tensor_builder,
+                                           std::shared_ptr<ITensorRegistry> &tensor_registry)
+  : _tgraph(tgraph), _tensor_builder(tensor_builder)
+{
+  _tensor_reg = std::dynamic_pointer_cast<TensorRegistry>(tensor_registry);
+}
+
+void ExtraTensorGenerator::register_tensors(ir::OperationIndex op_idx, ExtraTensorRequests &&reqs)
+{
+  // save request, _idx_to_reuqests will be used for memory planning
+  if (reqs.size() == 0)
+    return;
+
+  // _idx_to_requests[op_idx] = reqs;
+  _idx_to_requests.insert({op_idx, reqs});
+  auto &operations = _tgraph->operations();
+
+  for (size_t i = 0; i < reqs.size(); i++)
+  {
+    // register tensor
+    ExtraTensorIndex tensor_idx(op_idx, i);
+    _tensor_builder->registerExtraTensorInfo(tensor_idx, reqs[i].info());
+
+    std::stringstream op_info;
+    op_info << op_idx << "_" << operations.at(op_idx).name();
+    VERBOSE(ExtraTensorGenerator) << "register (idx:" << tensor_idx << ") requested from "
+                                  << op_info.str() << std::endl;
+
+    // return registered tensor
+    auto generated_tensor = _tensor_reg->getExtraTensor(tensor_idx);
+    reqs[i].update_address(generated_tensor);
```

Comment:
`ExtraTensorRequests &&reqs` is updated, but the updated requests may be unused within the function and destroyed after returning the function. So, it seems that this update is meaningless.

Reply:
I understand why this function is called here. `update_address` sets `ExtraTensor` raw pointers in layers.

Reply:
I feel sorry to make you take some time here.

Reply:
> I feel sorry to make you take some time here.

Rather, I feel sorry I haven't been paying attention to this work for a while.